pkg/iam/oidc/api.go

```diff
@@ -180,6 +180,21 @@ func (m *OpenIDConnectManager) MakeAssumeRolePolicyDocumentWithServiceAccountCon
 	})
 }
 
+// MakeAssumeRolePolicyDocumentWithServiceAccountConditionsAllowingWildcard constructs a trust policy document
+// that allows wildcard pattern matching in the subject condition. The subjectPattern should be in the format
+// "system:serviceaccount:namespace:name-pattern" where name-pattern can include wildcards like "*".
+func (m *OpenIDConnectManager) MakeAssumeRolePolicyDocumentWithServiceAccountConditionsAllowingWildcard(serviceAccountNamespace, subjectPattern string) cft.MapOfInterfaces {
```

Comment:
Could we update ```MakeAssumeRolePolicyDocumentWithServiceAccountConditions``` to take `subjectPattern` instead of creating another function to avoid duplicating some of the logic?